lib/improver/utilities/temporal_interpolation.py

```diff
@@ -149,76 +149,51 @@ def construct_time_list(self, initial_time, final_time):
-            CoordinateNotFoundError : If original cube does not have the
-                coordinate in dim_coords.
-            CoordinateNotFoundError : If coordinate in original cube is
-                not in new_cube.
         """
-        if dim_coords is None:
-            coord_list = [coord.name() for coord in cube.coords()]
-        else:
-            coord_list = []
-            for coord_name in dim_coords:
-                try:
-                    req_dim = cube.coord_dims(coord_name)
-                    associated_coords = [crd.name() for crd in cube.coords()
-                                         if cube.coord_dims(crd) == req_dim
-                                         and crd.name() is not coord_name]
-                    coord_list.append(coord_name)
-                    for val in associated_coords:
-                        coord_list.append(val)
-                except CoordinateNotFoundError:
-                    msg = ('Original cube does not have the coordinate in'
-                           ' dim_coords {}'.format(coord_name))
-                    raise CoordinateNotFoundError(msg)
-        for coord_name in coord_list:
-            dtype_orig = cube.coord(coord_name).points.dtype
-            try:
-                new_coord = new_cube.coord(coord_name)
-                # TODO: Provide safe support for time coordinates that could
-                # potentially be in units of "hours since 1970-01-01 00:00:00."
-                if new_coord.points.dtype != dtype_orig:
-                    new_coord.convert_units(
-                        "seconds since 1970-01-01 00:00:00")
-                    new_coord.points = np.around(new_coord.points)
-                    new_coord.points = new_coord.points.astype(dtype_orig)
-                if (hasattr(new_coord, "bounds")
-                        and new_coord.bounds is not None and
-                        new_coord.bounds.dtype != dtype_orig):
-                    new_coord.convert_units(
-                        "seconds since 1970-01-01 00:00:00")
-                    new_coord.bounds = np.around(new_coord.bounds)
-                    new_coord.bounds = new_coord.bounds.astype(dtype_orig)
-            except CoordinateNotFoundError:
-                msg = ('new_cube does not have the coordinate in'
-                       ' the original cube {}'.format(coord_name))
-                raise CoordinateNotFoundError(msg)
-        return new_cube
+        coord_dtypes = {
+            'time': np.int64,
+            'forecast_reference_time': np.int64,
+            'forecast_period': np.int32,
+            }
+        coord_units = {
+            'time': "seconds since 1970-01-01 00:00:00",
+            'forecast_reference_time': "seconds since 1970-01-01 00:00:00",
+            'forecast_period': "seconds",
+            }
+
+        for coord_name in ["time", "forecast_reference_time",
+                           "forecast_period"]:
+            coord = cube.coord(coord_name)
```

Comment:
What happens if coordinate is not on cube?  It probably will be in this case, so possibly OK for a plugin method, but wouldn't be OK for a general utility.

Reply:
I've added a check, so that the the datatype and unit will only be modified if the coordinate exist on the cube.

Reply:
Thanks, that works.